### ISSUE TITLE
Contents item on hyperlinks are skipped

### ISSUE THREAD

**Issue by FvdGDTP** [nvaccess/nvda#13982]
### Steps to reproduce:
Hyperlinks are correctly formatted with the description text in the Contents item key. InDesign 2022 writes this contents item key correctly. It should be read, instead NVDA only reads Al-text, that is not conform the pdf/ua standard: it should read the Contents item instead

### Actual behavior:
Skips the description in the Contents item key.

### Expected behavior:
To read the content in the Contents item key
-->

### System configuration
Windows
#### NVDA version:
Latest public release

#### Windows version:
11

#### Name and version of other software in use when reproducing the issue:
Acrobat Pro for PDF

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No, do previous versions work?

**Comment by seanbudd** [comment]
@FvdGDTP - Could you provide a sample PDF?
Just to confirm, this issue happens when using Adobe Acrobat?

**Comment by FvdGDTP** [comment]
Hello,
Will do at the end of the day when I’m back ;-)


met vriendelijke groet,

Frans van der Geest


---
***@***.***

**Comment by FvdGDTP** [comment]
Hello Sean,

here is a classic, it tests several things. But Contents item goes wrong:



regards,

F. van der Geest
---
***@***.***

**Comment by FvdGDTP** [comment]
>> @FvdGDTP <https://github.com/FvdGDTP> - Could you provide a sample PDF?
>> Just to confirm, this issue happens when using Adobe Acrobat?
>> 
>> 

Yes, Acrobat (Pro) DC, latest

regards,

F. van der Geest
---

**Comment by FvdGDTP** [comment]
<img width="697" alt="Schermafbeelding 2022-08-04 om 17 54 08" src="https://user-images.githubusercontent.com/26364276/182892538-f9f30360-0ad2-41a4-8e3d-cabd55451592.png">
[Actual-Alt-Expansion-Text.pdf](https://github.com/nvaccess/nvda/files/9261183/Actual-Alt-Expansion-Text.pdf)

**Comment by FvdGDTP** [comment]
[Actual-Alt-Expansion-Text.pdf](https://github.com/nvaccess/nvda/files/9261185/Actual-Alt-Expansion-Text.pdf)

**Comment by FvdGDTP** [comment]
The attached PDF example should read what is in the Contents item key as shown on the screenshot, but it skips it.

**Comment by FvdGDTP** [comment]
Any progress on this? It bites us all…

**Comment by Adriani90** [comment]
@FvdGDTP here are many blind developers who cannot read what is on the screenshot. Can you please write down exactly what you expect to hear from NVDA? What exactly is skipped here?

**Comment by FvdGDTP** [comment]
﻿Hi,Well, you have an Alt text field for Figures, an Contents item (or Alt text for Hyperlinks). Both are set in a different field in Acrobat and are coded somewhere else in the code in PDF.NVDA does read what is in the Alt text field for images, it does not read what is in a Alt text for hyperlinks field, the so called Contents item, it skips that.Furthermore: Actual text for an image should not announce ‘Figure’, it still does (in Dutch anyway that is).Alt text, Contents item, Actual text are three different things, three different fields in the PDF code, NVDA only reads Alt text for images as it should.Hope this helps.Best regards,FransVerstuurd vanaf mijn iPhoneVerstuurd vanaf mijn iPhoneOp 7 dec 2024 om 23:08 heeft Adriani90 ***@***.***> het volgende geschreven:﻿
@FvdGDTP here are many blind developers who cannot read what is on the screenshot. Can you please write down exactly what you expect to hear from NVDA? What exactly is skipped here?

**Comment by FvdGDTP** [comment]
Hi,
Well, you have an Alt text field for Figures, an Contents item (or Alt text for Hyperlinks). Both are set in a different field in Acrobat and are coded somewhere else in the code in PDF.
NVDA does read what is in the Alt text field for images, it does not read what is in a Alt text for hyperlinks field, the so called Contents item, it skips that.
Furthermore: Actual text for an image should not announce ‘Figure’, it still does (in Dutch anyway that is).
Alt text, Contents item, Actual text are three different things, three different fields in the PDF code, NVDA only reads Alt text for images as it should.
Hope this helps.
